micro_sam/util.py

```diff
@@ -203,14 +202,18 @@ def get_sam_model(
     Returns:
         The segment anything predictor.
     """
-    checkpoint = _get_checkpoint(model_type, checkpoint_path)
     device = _get_device(device)
 
+    if checkpoint_path is None:
+        checkpoint = models().fetch(model_type)
+    else:
+        checkpoint = _get_checkpoint(model_type, checkpoint_path)
+
     # Our custom model types have a suffix "_...". This suffix needs to be stripped
     # before calling sam_model_registry.
     model_type_ = model_type[:5]
     assert model_type_ in ("vit_h", "vit_b", "vit_l", "vit_t")
-    if model_type == "vit_t" and not VIT_T_SUPPORT:
+    if model_type_ == "vit_t" and not VIT_T_SUPPORT:
```

Comment:
It seems possible that in the future people might fine tune models based off `vit_t`, so we'd have `model_type="vit_t_em"`, etc. 
I find the `model_type`/`model_type_` distinction here confusing, I'd kinda like to rename `model_type` -> `model_name` instead.

Reply:
I fully agree, we should have a better distinction here and the current distinction is confusing. Let's do what you suggest and rename `model_type -> model_name` and `model_type_ -> model_type`